### PR TITLE
parsers: Add FHIR Shorthand (FSH)

### PR DESCRIPTION
**Pull request by mgramigna**
Language Info: https://build.fhir.org/ig/HL7/fhir-shorthand/overview.html
TS Grammar: https://github.com/mgramigna/tree-sitter-fsh

Developed the grammar alongside some developers of the language itself. It seems to work/be correct! Let me know if I did this contribution right.. first time contributing to `nvim-treesitter`. Thanks :)

### CODE REVIEW

**Review by ObserverOfTime — COMMENTED** [review]
Can some of the keywords use more specific captures?
(e.g. `@keyword.operator`, `@function`, `@type.qualifier`)

**Comment by mgramigna** [comment]
@ObserverOfTime Thanks for the review! I read through [the available highlight captures](https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md#highlights) some more, and updated some captures to use `@type.definition`, `@keyword.operator`, and `@type.builtin`, as those were more appropriate based on the description.

File: queries/fsh/highlights.scm [hunk 1]
```diff
@@ -29,7 +29,7 @@
   "Logical"
   "Resource"
   "RuleSet"
-] @keyword
+] @type.definition
```
Comment (ObserverOfTime) [inline queries/fsh/highlights.scm:32]: `@type.definition` is only used for type aliases.

**Review by ObserverOfTime — CHANGES_REQUESTED** [review]

File: queries/fsh/highlights.scm (hunk 1 above)
Comment (mgramigna) [reply queries/fsh/highlights.scm:32]: Ah okay, maybe type.builtin makes more sense for this one

Comment (ObserverOfTime) [reply queries/fsh/highlights.scm]: After a quick look at the docs, keeping it as `@keyword` is probably the best option.

Comment (mgramigna) [reply queries/fsh/highlights.scm]: Got it

**Review by theHamsta — APPROVED** [review]

**Pull request merged**